### PR TITLE
[v7r3] fix: from 50 to 200MB of buffer limit

### PR DESCRIPTION
**Pull request by fstagni**
BEGINRELEASENOTES

*Core
CHANGE: Subprocess: from 5 to 20MB of buffer limit

ENDRELEASENOTES

### CODE REVIEW

File: src/DIRAC/Core/Utilities/Subprocess.py [hunk 1]
```diff
@@ -575,7 +575,7 @@ def systemCall(timeout, cmdSeq, callbackFunction=None, env=None, bufferLimit=524
     return result
 
 
-def shellCall(timeout, cmdSeq, callbackFunction=None, env=None, bufferLimit=52428800):
+def shellCall(timeout, cmdSeq, callbackFunction=None, env=None, bufferLimit=20971520):
```
Comment (andresailer) [inline src/DIRAC/Core/Utilities/Subprocess.py]: This number is clearly smaller than before

Comment (chrisburr) [reply src/DIRAC/Core/Utilities/Subprocess.py]: https://github.com/DIRACGrid/DIRAC/commit/a9e9205af26ee19aea781bbdbe587fa92b4a70e6

File: src/DIRAC/Core/Utilities/Subprocess.py [hunk 2]
```diff
@@ -153,17 +153,17 @@ class Subprocess(object):
...
         :param int timeout: timeout in seconds
-        :param int bufferLimit: buffer size, default 5MB
+        :param int bufferLimit: buffer size, default 20MB
         """
         self.log = gLogger.getSubLogger("Subprocess")
         self.timeout = False
         try:
             self.changeTimeout(timeout)
-            self.bufferLimit = int(bufferLimit)  # 5MB limit for data
+            self.bufferLimit = int(bufferLimit)  # 20MB limit for data
```
Comment (andresailer) [inline src/DIRAC/Core/Utilities/Subprocess.py]: remove this comment here, so there is just one place that needs updating
```suggestion
            self.bufferLimit = int(bufferLimit)
```

File: src/DIRAC/Core/Utilities/Subprocess.py (hunk 1 above)
Comment (fstagni) [reply src/DIRAC/Core/Utilities/Subprocess.py]: Oh, so now I should even review my own PRs? 😠 

😄 OK upping to 200 MB

Comment (andresailer) [reply src/DIRAC/Core/Utilities/Subprocess.py]: `git add --patch` is the first line of review :exclamation: :smiling_face_with_tear:

File: src/DIRAC/Core/Utilities/Subprocess.py [hunk 3]
```diff
@@ -153,17 +153,17 @@ class Subprocess(object):
...
         :param int timeout: timeout in seconds
-        :param int bufferLimit: buffer size, default 5MB
+        :param int bufferLimit: buffer size, default 200MB
         """
         self.log = gLogger.getSubLogger("Subprocess")
         self.timeout = False
         try:
             self.changeTimeout(timeout)
-            self.bufferLimit = int(bufferLimit)  # 5MB limit for data
+            self.bufferLimit = int(bufferLimit)  # 200MB limit for data
```
Comment (andresailer) [inline src/DIRAC/Core/Utilities/Subprocess.py:166]:
```suggestion
            self.bufferLimit = int(bufferLimit)
```

**Comment by chrisburr** [comment]
I think 200MB is too high and this should be fixed properly by modifying the JobAgent to not use the DIRAC Subprocess module and instead stream the output to disk (like is done in LHCbDIRAC's RunApplication).

**Comment by fstagni** [comment]
> I think 200MB is too high and this should be fixed properly by modifying the JobAgent to not use the DIRAC Subprocess module and instead stream the output to disk (like is done in LHCbDIRAC's RunApplication).

That would be better... but I said I didn't want to think on this one ! :)

anyway, that would require some `async` which I am not familiar with and that would need to go in a non-production branch. 

So, we can close this PR and either create an issue from this comment, or you take care :)

**Comment by fstagni** [comment]
The issue is already in #5575

**Pull request closed without merging**